homeassistant/components/spotify/media_player.py

```diff
@@ -299,6 +300,13 @@ def play_media(self, media_type: str, media_id: str, **kwargs) -> None:
             kwargs["uris"] = [media_id]
         elif media_type == MEDIA_TYPE_PLAYLIST:
             kwargs["context_uri"] = media_id
+            if self.shuffle and "playlist" in media_id:
+                response = self._spotify.playlist_tracks(
+                    media_id, offset=0, fields="total"
+                )
+                song_count = response.get("total", 1)
+                position = randint(0, song_count - 1)
```

Comment:
Normally we don't allow adding fake features to device integrations. Our standard is to only expose features that the device api exposes.

Reply:
but this was working in HomeAssistant 104.x, version 105.x broke this capability

Reply:
If the device library has support for this feature we should try to fix the device library or the implementation of that library here. We shouldn't add a workaround.

Reply:
@starkillerOG This workaround/feature was present in the old/previous implementation. Just because we allowed for more "bad" design decisions (please no, I mean no harm, hence the quotes) in the past, doesn't mean we should make them again.

Reply:
@frenck you have a point.
Just a bit frustrated because I really want to fix this issue and finish it.